lib/honeybadger/config/ruby.rb

```diff
@@ -87,14 +87,12 @@ def backtrace_filter
       end
 
       def before_notify(action = nil, &block)
-        hash[:before_notify] ||= []
-
-        if action
-          hash[:before_notify] << action
-        elsif block_given?
-          hash[:before_notify] << block
-        else
-          get(:before_notify)
+        (hash[:before_notify] ||= []).tap do |before_notify_hooks|
+          if action
+            before_notify_hooks << action if validate_before_action(action)
```

Comment:
A minor nitpick, can we move this if to the previous line make it `if action && validate_before_action(action)` and the same for line:93

Reply:
Great suggestion!